core/src/test/java/org/apache/iceberg/TestMetadataTableScansWithPartitionEvolution.java

```diff
@@ -243,6 +245,102 @@ public void testPartitionSpecEvolutionToUnpartitioned() throws IOException {
     }
   }
 
+  @TestTemplate
+  public void testPartitionSpecEvolutionNullValues() throws IOException {
+    Schema schema =
+        new Schema(
+            required(1, "company_id", Types.IntegerType.get()),
+            required(2, "dept_id", Types.IntegerType.get()),
+            required(3, "team_id", Types.IntegerType.get()));
+
+    table =
+        TestTables.create(
+            tableDir,
+            metadataDir,
+            "nulltest",
+            schema,
+            PartitionSpec.builderFor(schema).identity("company_id").build(),
+            SortOrder.unsorted(),
+            formatVersion);
+    table.newFastAppend().appendFile(newDataFile("company_id=__HIVE_DEFAULT_PARTITION__")).commit();
+
+    table.updateSpec().addField("dept_id").commit();
+    table
+        .newFastAppend()
+        .appendFile(
+            newDataFile(
+                "company_id=__HIVE_DEFAULT_PARTITION__" + "/dept_id=__HIVE_DEFAULT_PARTITION__"))
+        .commit();
+
+    table.updateSpec().addField("team_id").commit();
+    table
+        .newFastAppend()
+        .appendFile(
+            newDataFile(
+                "company_id=__HIVE_DEFAULT_PARTITION__"
```

Comment:
Is there a reason to do this with partition paths instead of structs? I generally would like to avoid using strings if at all possible

Reply:
no reason, I followed the style used in this test class: https://github.com/apache/iceberg/pull/12528/files#diff-31a7139557d5a20af8245ddf3f9a4ff125094cd5d4f50b19e08104fecbabf98dR64-R66

Reply:
note, this test class uses TestTables.TestTable wrapper that doesn't support `StructLike partition`

Reply:
refactored to use struct